fenwick/src/main/java/org/opencadc/fenwick/Main.java

```diff
@@ -1,114 +1,247 @@
+            }
+
             InventoryHarvester doit = new InventoryHarvester(daoConfig, resourceID, selector);
             doit.run();
             System.exit(0);
         } catch (Throwable unexpected) {
-            log.error("unexpected failure", unexpected);
+            log.error("Unexpected failure", unexpected);
+            System.err.println("Unexpected failure.  Check log output.");
             System.exit(-1);
         }
     }
-    
-    private Main() { 
+
+    /**
+     * Read in the configuration file and load it into the given Properties.
+     */
+    private static void configure(final MultiValuedProperties properties) throws IOException {
+        // Read in the configuration file, and create a usable Map from it.
+        final String homeConfigDirectoryPath = String.format("%s/config", System.getProperty("user.home"));
+
+        // Allow some flexibility to override.  This is useful for local testing as one would normally require root
+        // access to create /config.
+        final String configurationDirectory = new File(homeConfigDirectoryPath).isDirectory()
+                                              ? homeConfigDirectoryPath : "/config";
+
+        final String configurationFileLocation = String.format("%s/%s", configurationDirectory, CONFIG_FILE_NAME);
+
+        final InputStream configFileInputStream = new FileInputStream(configurationFileLocation);
+        properties.load(configFileInputStream);
+    }
+
+    /**
+     * Convenience method to acquire required property values, or append to an error message builder if no property
+     * exists for the given key.
+     *
+     * @param properties          The properties to read from.
+     * @param key                 The key to look up.
+     * @param errorMessageBuilder The error message builder to append to.  Can be null.
+     * @return Value of the property, or null if non existent.
+     */
+    private static String readProperty(final MultiValuedProperties properties, final String key,
+                                       final StringBuilder errorMessageBuilder) {
+        final String configuredValue = properties.getFirstPropertyValue(key);
+        final String returnValue;
+        if (!StringUtil.hasText(configuredValue)) {
+            if (errorMessageBuilder != null) {
+                errorMessageBuilder.append("\n").append(key);
+            }
+            returnValue = null;
+        } else {
+            returnValue = configuredValue;
+        }
+
+        return returnValue;
+    }
+
+    /**
+     * Set any package's logging level here.  Allows for finer grained messages.
+     *
+     * @param applicationProperties The properties read from a file.
+     */
+    private static void setLogging(final MultiValuedProperties applicationProperties) {
+        Log4jInit.setLevel("ca.nrc.cadc", Level.WARN);
+        Log4jInit.setLevel("org.opencadc", Level.WARN);
+
+        final String loggingKey = ".logging";
+
```

Comment:
Why is there a loop here? The README says the value of the logging prop is a single "log level" value. We decided to hard code packages that get that level.